client/pages/sign-up.jsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import { Form, Button } from 'react-bootstrap';
+
+export default class SignUpForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      email: '',
+      password: '',
+      passwordVerfiy: '',
+      latitude: null,
+      longitude: null,
+      location: ''
+    };
+    this.handleUsername = this.handleUsername.bind(this);
+    this.handleEmail = this.handleEmail.bind(this);
+    this.handlePassword = this.handlePassword.bind(this);
+    this.handlePasswordVerify = this.handlePasswordVerify.bind(this);
+    this.handleLocation = this.handleLocation.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+
+    window.navigator.geolocation.getCurrentPosition(position => {
+
+      fetch(`https://maps.googleapis.com/maps/api/geocode/json?latlng=${position.coords.latitude},${position.coords.longitude}&key=${process.env.GOOGLE_MAPS_API_KEY}`)
+        .then(res => res.json())
+        .then(currentLocation => {
+          for (let i = 0; i < currentLocation.results.length; i++) {
+            const result = currentLocation.results[i];
+            if (result.types.includes('postal_code')) {
+
+              this.setState({
+                latitude: position.coords.latitude,
+                longitude: position.coords.longitude,
+                location: result.formatted_address
+              });
+            }
```

Comment:
It would also be good to include `break` after `this.setState()` so that the loop doesn't keep going